### ISSUE TITLE
Components parameters are shared/public between instances

### ISSUE THREAD

**Issue by joao-oliveira-nex** [EFTEC/BladeOne#131]
When I create a component and I invoke it multiple times, if an optional parameter is not passed, it will assume the last passed value for that parameter on another instance of that component.

Take my example. This is my component for importing icons, it has the optional variables $size and $color which should be rendered as 'false' by default:
```
<span class="ico ico--{{$size or 'false'}} ico--{{$color or 'false'}}">
  {!! file_get_contents($manifest['/icons/ico-' . $icon . '.svg']) !!}
</span>
```

I invoke it for the first time on the header with an 'sm' size and a 'white' color. Like the following code:
```
@component('components.icon.icon', ['icon' => 'user', 'size' => 'sm', 'color' => 'white'])
@endcomponent
```

And this will render properly:
```
<span class="ico ico--sm ico--white">
  <svg> ... user icon ... </svg>
</span>
```

But then I invoke another instance of the component on the footer, with another icon, but I want the default color and size:
```
@component('components.icon.icon', ['icon' => 'linkedin'])
@endcomponent
```

And what I get is, with the size and color variable values of the header import instance, as they are not passed:
```
<span class="ico ico--sm ico--white">
  <svg> ... linkedin icon ... </svg>
</span>
```

What I expected to get is:
```
<span class="ico ico--false ico--false">
  <svg> ... linkedin icon ... </svg>
</span>
```

I even tried using an @include method, but the behaviour is the same.
Is it supposed to be like this? I mean. This also makes the {{$size or 'false'}} variable calling kind of useless on the components and pretty much any default behaviour.

Thank you for your kind attention!
Kindest Regard.

**Comment by darkfire56j** [comment]
Same here, I suppose is matter of php scope, but I have no idea how to fix this issue. Can anyone suggest a way to solve?

**Comment by jorgecc** [comment]
I found the problem but I am a bit short of time solving and testing it. So, in January.

**Comment by jorgecc** [comment]
fixed in 3.48

If you find an error or mistake, then open a new issue.
